src/python/gem5/utils/looppoint.py

```diff
@@ -0,0 +1,404 @@
+
+    def get_current_region(self) -> int:
+        """
+        This function returns the region id if the current PC Count pair is
+        significant(e.x. beginning of the checkpoint), otherwise, it returns
+        a '-1' to indicate the current PC Count pair is not significant
+        """
+        current_pair = self._manager.get_current_pc_count_pair()
+        temp_pair = PcCountPair(current_pair.getPC(), current_pair.getCount())
+        if temp_pair in self._regions:
+            return self._regions[temp_pair]
+        return -1
+
+    def get_current_pair(self) -> PcCountPair:
+        """
+        This function returns the current PC Count pair
+        """
+        return self._manager.get_current_pc_count_pair()
+
+    def get_regions(self) -> Dict[PcCountPair, int]:
+        """
+        This function returns the complete dictionary of _regions
+        """
+        return self._regions
+
+    def get_targets(self) -> List[PcCountPair]:
+        """
+        This function returns the complete list of _targets
+        """
+        return self._targets
+
+
+class LoopPointCheckpoint(LoopPoint):
+    def __init__(self, looppoint_file: Path, if_csv: bool) -> None:
+        """
+        This class is specifically designed to take in the LoopPoint data file
+        and generate the information needed to take checkpoints for LoopPoint
+        regions(warmup region+simulation region)
+        :param looppoint_file: the director of the LoopPoint data file
+        :param if_csv: if the file is a csv file, then it is True. If the file
+        is a json file, then it is False
+        """
+
+        _json_file = {}
+        _targets = []
+        _region_id = {}
+
+        if if_csv:
+            self.profile_csv(looppoint_file, _targets, _json_file, _region_id)
+        else:
+            self.profile_json(looppoint_file, _targets, _json_file, _region_id)
+
+        super().__init__(
+            _targets,
+            _region_id,
+            _json_file,
+        )
+
+    def profile_csv(
+        self,
+        looppoint_file_path: Path,
+        targets: List[PcCountPair],
+        json_file: Dict[int, Dict],
+        region_id: Dict[PcCountPair, int],
+    ) -> None:
+        """
+        This function profiles the csv LoopPoint data file into three variables
+        to take correct checkpoints for LoopPoint
+        :param looppoint_file_path: the director of the LoopPoint data file
+        :param targets: a list of PcCountPair
+        :param json_file: a dictionary for all the LoopPoint data
+        :param region_id: a dictionary for all the significant PcCountPair and
+        its corresponding region id
+        """
+
+        # This section is hard-coded to parse the data in the csv file.
+        # The csv file is assumed to have a constant format.
+        with open(looppoint_file_path, newline="") as csvfile:
+            spamreader = csv.reader(csvfile, delimiter=" ", quotechar="|")
```

Comment:
needs a better name than `spamreader`. `f` or `reader` or `data` would all be better